core/win32/drwinapi/kernel32_proc.c

```diff
@@ -48,8 +48,11 @@ void
 kernel32_redir_init_proc(void)
 {
     PEB *peb = get_own_peb();
+    /* FIXME i#3633: Implement FLS isolation in Win10 1903+ where FLS data is no longer
```

Comment:
We have an implementation so no need for the word FIXME, right?  If you want to talk about ways to make the implementation here better feel free, though I would that would be XXX since it's a nice-to-have not a must-have, and the comment should spell out what we'd do differently.

Reply:
I could write on this one a bit more, but I at the moment I think it is good to go for 1903. Don't know if DynamoRIO gives ability to intercept APIs from the engine itself - `drwrap_wrap` like functionality. With this, it would be trivial to intercept FLS APIs, and properly redirect them, or handle FLS without ntdll.dll ever touching it. Maybe for the future?

Reply:
drwrap is for intercepting the app, which is run under DR control (or possibly natively for the not-really-supported Probe API).  Here we need to intercept libraries running natively in tool mode.  The private loader can do this at the inter-library import level, as is done with the Fls functions in drwinapi/ today.